geoh5py/data/numeric_data.py

```diff
@@ -76,7 +76,11 @@ def format_length(self, values: np.ndarray) -> np.ndarray:
         :param values: the values to check.
         :return: the values with the right length.
         """
-        if self.n_values is not None:
+
+        if (
+            not any(k in self.name for k in ["DEPTH", "FROM", "TO"])
```

Comment:
Would be better if the check happened in the concatenated data, so that we don't cluter the base class with drillhole stuff. Maybe `return self.n_values = None` if it's a DEPTH, FROM or TO data, then we just return the value here without extending.

Reply:
Done